src/Data/Foldable.js

```diff
@@ -0,0 +1,26 @@
+/* global exports */
+"use strict";
+
+// module Data.Foldable
+
+exports.foldrArray = function (f) {
+  return function (init) {
+    return function (xs) {
+      /* jshint maxparams: 2 */
+      return xs.reduceRight(function (acc, x) {
```

Comment:
Perhaps the jsperf one is faster because it can use the mutating assignment `+=`?